### PR TITLE
Closes #301: Cleanup recovery flow

### LINKED ISSUE #301: Clean up recovery flow with additional check
Speak to @MakotoTheKnight for more details.

### PR DESCRIPTION
**Pull request by MakotoTheKnight**
This PR is good to go.  For details on how to test it, please see @MakotoTheKnight.

### CODE REVIEW

File: spec/controllers/v1/user_controller_spec.rb [hunk 1]
```diff
@@ -184,42 +188,122 @@ def post_create_user(options = {})
       kim = User.find_by(email: 'kim@gov.nk')
       expect(kim.organizations.count).to eq(2)
     end
-    
+
     it 'can find and update an existing user' do
```
Comment (juliosantos) [inline spec/controllers/v1/user_controller_spec.rb]: Shouldn't this be "can't"?

File: app/controllers/v1/user_controller.rb [hunk 2]
```diff
@@ -3,27 +3,34 @@ class V1::UserController < ApplicationController
...
 
-    user_invitation = [UserInvitation, InventoryInvitation].map{|i| i.find_by(email: user_params[:email])}.compact.first
-    if user_invitation.present?
-      @user.errors.add(:base, "It seems you have already been invited. Please continue your registration here.")
+    user_invitation = [UserInvitation, InventoryInvitation].map { |i|
+      i.find_by(email: user_params[:email])
+    }.compact.first
+    invite_user = user_invitation.try(:user)
+
+    if invite_user && invite_user.sign_in_count > 0
```
Comment (juliosantos) [inline app/controllers/v1/user_controller.rb]: After a user redeems an invitation, `invite_user` should be `nil`. I don't see how this branch is ever reached.

**Comment by juliosantos** [comment]
It could be that I'm misunderstanding the scope, but this still allows impersonating someone that hasn't redeemed their invitation yet.

**Comment by MakotoTheKnight** [comment]
@juliosantos:  You're right.  This PR wasn't intended to address that aspect of it.  This PR was intended to guard against a user with an existing invitation in the system who had previously signed in, but did not have their invitation deleted.  Effectively, this doesn't work for brand new accounts, but it does work for the legacy accounts in our system, of which we've got ~192.

I'd almost prefer to investigate how it's possible that accounts got into that state prior to the day we implemented the fix, but that's not entirely germane to guarding against this possibility.

File: app/controllers/v1/user_controller.rb (hunk 2 above)
Comment (MakotoTheKnight) [inline app/controllers/v1/user_controller.rb]: I've added some notes and did a bit of looking around.  We have the existing state in our code base in that there are still some invitations lurking about, and there are accounts with a lot of sessions logged (some over 100).

**Comment by MakotoTheKnight** [comment]
I will be tentatively merging this work in. There will be instructions on how to test it in our Slack channel.

**Pull request merged**